addons/crm/models/crm_lead.py

```diff
@@ -551,6 +552,12 @@ def _compute_partner_phone_update(self):
         for lead in self:
             lead.partner_phone_update = lead._get_partner_phone_update()
 
+    @api.depends_context('uid')
+    @api.depends('partner_id')
+    def _compute_is_partner_visible(self):
```

Comment:
Could add a bit of explanation here in the docstring.

```
""" When the crm.lead is of type 'lead', we don't want to display the "Customer" field on the form view
unless it's set (or debug mode).

Indeed, most of the times leads will not have this information set, since when we assign a Customer we
usually convert the lead to an opportunity as well.

This means that on the lead form, we don't want to display this field since it may be misleading for the
end user.
When it's set however, we want to display it, mainly because there are a few automatic synchronizations between
the lead and its partner (phone and email for examples), and this needs to be clear that modifying
one of those fields will in turn modify the linked partner. """
```

This explanation should be in the commit message as well IMO.